py/desispec/scripts/zcatalog_wrap.py

```diff
@@ -0,0 +1,226 @@
+            help="output directory")
+    parser.add_argument("--header", type=str, default=None,
+            help="KEYWORD=VALUE entries to add to the output header")
+    # parser.add_argument("--survey", type=str, nargs="*", default=None,
+    #         help="DESI survey, e.g. sv1, sv3, main")
+
+    parser.add_argument('--nproc', type=int, default=1,
+            help="Number of multiprocessing processes to use")
+
+    parser.add_argument("--minimal", action='store_true',
+            help="only include minimal output columns")
+    parser.add_argument('--patch-missing-ivar-w12', action='store_true',
+            help="Use target files to patch missing FLUX_IVAR_W1/W2 values")
+    parser.add_argument('--recoadd-fibermap', action='store_true',
+            help="Re-coadd FIBERMAP from spectra files")
+    parser.add_argument('--add-units', action='store_true',
+            help="Add units to output catalog from desidatamodel "
+                 "column descriptions")
+    parser.add_argument('-v', '--verbose', action='store_true',
+                        help="Set log level to DEBUG.")
+    args = parser.parse_args(options)
+
+    return args
+
+
+def main(args=None):
+    if not isinstance(args, argparse.Namespace):
+        args = parse(options=args)
+
+    if args.verbose:
+        log=get_logger(DEBUG)
+    else:
+        log=get_logger()
+
+    ## Ensure we know where to save the files
+    if args.outdir is None and args.cat_version is None:
+        log.critical(f"Either --outdir or --cat-version must be specifiied. Exiting")
+        return 1
+
+    ## If adding units, check dependencies before doing a lot of work
+    if args.add_units:
+        try:
+            import desidatamodel
+        except ImportError:
+            log.critical('Unable to import desidatamodel, required to add units (try "module load desidatamodel" first)')
+            return 1
+
+    ## Define filetype based on healpix vs not healpix
+    if args.group == 'healpix':
+        ftype = 'zcat_hp'
+    else:
+        ftype = 'zcat_tile'
+
+    ## Ensure input directory exists
+    if args.indir is not None and not os.path.exists(args.indir):
+        log.critical(f"Input directory {args.indir} does not exist.")
+        return 1
+
+    if args.outdir is None:
+        ## since we only care about the directory path, we can
+        ## use dummy values for survey and faprogram
+        args.outdir = os.path.dirname(findfile(ftype,
+                                               version=args.cat_version,
+                                               groupname=args.group,
+                                               survey='dummy', faprogram='dummy'))
+
+    ## Define catalog version (currently not used except with findfile,
+    ## but since we are only using findfile for the filename and not the
+    ## directory tree, the version isn't actually relevant
+    if args.cat_version is None:
+        args.cat_version = os.path.basename(args.outdir)
```

Comment:
This default `--cat-version` logic is a big subtle, turning the basename into the catalog version which then propagates into the output filenames.  Consider
* making `--cat-version` a required parameter (a slight hassle but safer, and potentially mitigated by making a `-V` shortcut, or `-v` and making `--verbose` be the long thing without a `-v` shortcut because it is afterall, verbose, like this comment...); or
* logging a warning message about how `cat_version` is being derived given that it wasn't specified on the command line.

Reply:
I've added a -V shortcut and made it required.